lib/manager/npm/post-update/lerna.ts

```diff
@@ -13,13 +12,28 @@ export interface GenerateLockFileResult {
   stderr?: string;
 }
 
+// Exported for testability
+export function getLernaVersion(
+  lernaPackageFile: Partial<PackageFile>
+): string {
+  const lernaDep = lernaPackageFile.deps?.find((d) => d.depName === 'lerna');
+  if (!lernaDep || !semver.validRange(lernaDep.currentValue)) {
+    logger.warn(
+      `Could not detect lerna version in ${lernaPackageFile.packageFile}, using 'latest'`
+    );
+    return 'latest';
+  }
+  return lernaDep.currentValue;
+}
+
 export async function generateLockFiles(
-  lernaClient: string,
+  lernaPackageFile: Partial<PackageFile>,
```

Comment:
Why do we use `Partial` here?

Reply:
That's what https://github.com/renovatebot/renovate/blob/master/lib/manager/npm/post-update/index.ts#L637 is passing to this function -- is there a way to assert that the given object is _not_ partial and unwrap it?

I do not know why `AdditionalPackageFiles.npm` is an array of `Partial<PackageFile>` however. (since PackageFile's properties are all (?) optional, this seems indeed perhaps unnecessary?